src/ExternalSources/ExternalSourceBase.cs

```diff
@@ -0,0 +1,42 @@
+namespace Flamenco.ExternalSources;
+
+public abstract class ExternalSourceBase
+{
+    public abstract Task Download(string destinationDirectory);
+
+    public static ExternalSourceBase Create(string descriptorFilePath)
+    {
+        if (!File.Exists(descriptorFilePath))
+        {
+            throw new FileNotFoundException("Descriptor file not found.", descriptorFilePath);
+        }
+
+        var fileContentLines = File.ReadAllLines(descriptorFilePath);
+        var type = fileContentLines.FirstOrDefault(l => l.StartsWith("type"))?.Split("=").Last();
+
+        if (type is null)
+        {
+            throw new ApplicationException($"{descriptorFilePath} is missing a type.");
+        }
+
+        switch (type)
+        {
+            case "git":
+                var repositoryUrl = fileContentLines.FirstOrDefault(l => l.StartsWith("repo"))?.Split('=').Last();
+                var branch = fileContentLines.FirstOrDefault(l => l.StartsWith("branch"))?.Split('=').Last();
+                var tag = fileContentLines.FirstOrDefault(l => l.StartsWith("tag"))?.Split('=').Last();
+                var commit = fileContentLines.FirstOrDefault(l => l.StartsWith("commit"))?.Split('=').Last();
```

Comment:
question: Why do you use a custom file format instead of established formats like json, yaml or even xml?

Reply:
It's an [INI](https://en.wikipedia.org/wiki/INI_file)-style file. I aimed at a less verbose, to-the-point, and easily parse-able format. JSON is an option, although INI is even less verbose than that. But I don't think YAML or even XML are suitable, due to their unfriendliness.